### PR TITLE
PR 3

### PR DESCRIPTION
**Pull request by jokano12**
References: (8)
PhoneDataset
PhoneSaveData
EventData
MembershipData
CustomFieldData
CommitteeMemberData
RegistrationData
RelationshipData

Actions: (10)
Get All Events
Get Organization Active Memberships
Get Organization Custom Field Values
Get Committee Information for an Individual
Add Phone to Individual
Get Individual Active Memberships
Get All Event Registrations Information for an Individual
Get Individual's Relationships
Update an Individual Email
Save Relationship for Organization

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 1]
```diff
@@ -3985,418 +4992,1389 @@
...
+                "type": "integer",
+                "description": "Type.",
+                "title": "Type"
+            },
+            "country": {
+                "type": "object",
+                "properties": {
+                    "id": {
+                        "type": "string",
+                        "description": "Country ID.",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Country -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 2]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "title": "Country ID"
+                    },
+                    "name": {
+                        "type": "string",
+                        "description": "Country Name.",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Country -` prefix here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 3]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "title": "Country Name"
+                    },
+                    "twoLetterIsoCode": {
+                        "type": "string",
+                        "description": "Two Letter ISO Code.",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Country -` prefix here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 4]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "title": "Two Letter ISO Code"
+                    },
+                    "threeLetterIsoCode": {
+                        "type": "string",
+                        "description": "Three Letter ISO Code.",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Country -` prefix here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 5]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "title": "Three Letter ISO Code"
+                    },
+                    "code": {
+                        "type": "integer",
+                        "description": "Country Code.",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Country -` prefix here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 6]
```diff
@@ -3985,418 +4992,1389 @@
...
-          "200": {
-            "description": "Success",
-            "schema": {}
-          }
+    },
+    "CommitteeMemberData": {
+        "type": "object",
+        "properties": {
+            "committee": {
+                "$ref": "#/definitions/CommitteeData"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please replace this with the content, adding the `Committe -` prefix to the descriptions

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 7]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "description": "Phones."
+                    },
+                    "webSite": {
+                        "type": "string",
+                        "description": "Web Site.",
+                        "title": "Web Site"
+                    },
+                    "memberships": {
+                        "type": "array",
+                        "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 8]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "description": "Memberships."
+                    },
+                    "category": {
+                        "type": "string",
+                        "description": "Category.",
+                        "title": "Category"
+                    },
+                    "categories": {
+                        "type": "array",
+                        "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 9]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "description": "Categories"
+                    },
+                    "customFields": {
+                        "type": "array",
+                        "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 10]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        "description": "Show In Directory.",
+                        "title": "Show In Directory",
+                        "enum": [
+                            true,
+                            false
+                        ]
+                    },
+                    "links": {
+                        "type": "array",
+                        "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 11]
```diff
@@ -3985,418 +4992,1389 @@
...
+                            "type": "string",
+                            "description": "Registration Date."
+                        },
+                        "externalCode": {
+                            "type": "string",
+                            "description": "External Code."
+                        },
+                        "customFields": {
+                            "type": "array",
+                            "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 12]
```diff
@@ -3985,418 +4992,1389 @@
...
+                            "description": "Additional Info."
+                        },
+                        "quantity": {
+                            "type": "integer",
+                            "format": "int32",
+                            "description": "Quantity."
+                        },
+                        "itemizedCustomFields": {
+                            "type": "array",
+                            "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 13]
```diff
@@ -3985,418 +4992,1389 @@
...
+                            "description": "Itemized Custom Fields."
+                        },
+                        "productCategories": {
+                            "type": "array",
+                            "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 14]
```diff
@@ -3985,418 +4992,1389 @@
...
-                "secret",
-                "typeName",
-                "description",
-                "filters"
-              ]
+                "description": "Related Products."
+            },
+            "tags": {
+                "type": "array",
+                "items": {},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add the definition of the array items here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 15]
```diff
@@ -3985,418 +4992,1389 @@
...
+                "items": {
+                    "type": "string"
                 },
-                "callBackUrl": {
-                  "type": "string",
-                  "description": "Callback URL.",
-                  "title": "Callback URL",
-                  "x-ms-visibility": "internal",
-                  "x-ms-notification-url": true
+                "description": "imageUrl"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please fix this description

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 16]
```diff
@@ -3985,418 +4992,1389 @@
...
+            "externalCode": {
+                "type": "string",
+                "description": "External Code.",
+                "title": "External Code"
+            },
+            "eventLocations": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6382]: Please add `Event Locations -` prefix to these properties
Also add missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 17]
```diff
@@ -3985,418 +4992,1389 @@
...
+                        },
+                        "description": {
+                            "type": "string",
+                            "description": "Description."
+                        },
+                        "addresses": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6398]: Please add `Addresses -` prefix to these properties
Also add missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 18]
```diff
@@ -3985,418 +4992,1389 @@
...
+                                        "type": "boolean",
+                                        "description": "Show In Directory."
+                                    },
+                                    "congressionalDistrict": {
+                                        "type": "string",
+                                        "description": "Congressional District."
+                                    },
+                                    "countryData": {
+                                        "type": "object",
+                                        "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6490]: Please add `Country -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 19]
```diff
@@ -3985,418 +4992,1389 @@
...
-                  "x-ms-visibility": "internal",
-                  "default": "Customer.RequestUpdated"
+                "description": "Event Locations."
+            },
+            "categories": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "$id": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please remove this property as it's not part of the response.
Check http://abc.mpxdemo.com/api/help/ResourceModel?modelName=CategoryData

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 20]
```diff
@@ -3985,418 +4992,1389 @@
...
-                  "title": "Type Name",
-                  "x-ms-visibility": "internal",
-                  "default": "Customer.RequestUpdated"
+                "description": "Event Locations."
+            },
+            "categories": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (LordLydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6083]: please add `Categories -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 21]
```diff
@@ -3985,418 +4992,1389 @@
...
-                    "default": "NA"
-                  },
-                  "description": "Filters."
+                "description": "Categories."
+            },
+            "educationCredits": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6605]: please add `Education Credits -` prefix to descriptions and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 22]
```diff
@@ -3985,418 +4992,1389 @@
...
-                  "title": "Secret",
-                  "x-ms-visibility": "internal",
-                  "default": "Impexium"
+                "description": "Education Credits."
+            },
+            "uploadsData": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6636]: Please add `Uploads -` prefix to description and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 23]
```diff
@@ -3985,418 +4992,1389 @@
...
-                  "x-ms-notification-url": true,
-                  "x-ms-visibility": "internal",
-                  "title": "Callback URL"
+                "description": "Uploads Data"
+            },
+            "relatedProducts": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6671]: Add `Related Products -` prefix to description and missing `title`s

**Review by lordlydon — CHANGES_REQUESTED** [review]

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 24]
```diff
@@ -3903,500 +4997,1741 @@
...
+                            "title": "Code"
+                        },
+                        "name": {
+                            "type": "string",
+                            "description": "Categories - Name",
+                            "title": "Name"                     
+                        },
+                        "isPrimary": {
+                            "type": "boolean",
+                            "description": "Categories - Is Primary."
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Missing comma `,`

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 25]
```diff
@@ -3903,500 +4997,1741 @@
...
+            },
+            "educationCredits": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "code": {
+                            "type": "string",
+                            "description": "Education Credits - Code.",
+                            "title": "Code
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Missing closing quotation mark `"`

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 26]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                    },
+                                    "isPrimary": {
+                                        "type": "boolean",
+                                        "description": "Is Primary."
+                                    }
+                                }
+                            },
+                            "description": "Sub Categories."
+                        }
+                            }
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Missing comma `,`

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 27]
```diff
@@ -3903,500 +4997,1741 @@
...
+                            "format": "int32",
+                            "description": "Quantity."
+                        },
+                        "itemizedCustomFields": {
+                            "type": "array",
+                            "items": {
+                              "type": "object",
+                              "properties": {
+                                "name": {"type": "string", "description": "Name."},
+                                "caption": {"type": "string", "description": "Caption".},
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: There is a dot `.` outside the quotation marks

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 28]
```diff
@@ -5278,7 +5278,87 @@
                     },
                     "memberships": {
                         "type": "array",
-                        "items": {},
+                        "items": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5321]: Please fix indentation

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 29]
```diff
@@ -5288,12 +5368,76 @@
...
+                            "type": "string",
+                            "description": "Name"
+                        },
+                        "isPrimary": {
+                            "type": "boolean",
+                            "description": "Is Primary."
+                        }
+                    }
+                },
+                "description": "Categories."
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: `description` and closing bracket `}` are duplicated here

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 30]
```diff
@@ -5288,12 +5368,76 @@
                     },
                     "categories": {
                         "type": "array",
-                        "items": {},
+                        "items": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5415]: `items`, and `type` are duplicated

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 31]
```diff
@@ -5701,12 +5859,50 @@
...
+                              "type": "object",
+                              "properties": {
+                                "code": {"type": "string", "description": "Code."},
+                                "name": {"type": "string", "description": "Name."},
+                                "IsPrimary": {"type": "boolean", "description": "Is Primary."},
+                                "subCategories": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Subcategories -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 32]
```diff
@@ -5278,7 +5278,87 @@
...
+            "membershipUniqueId": {
+                "type": "string",
+                "description": "Membership Unique ID.",
+                "title": "Membership Unique ID"
+            },
+            "termsList": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Terms -` prefix and add missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 33]
```diff
@@ -5288,12 +5368,76 @@
...
+                        "items": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "subCategories": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Subcategories -` prefix and missing titles

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 34]
```diff
@@ -3903,500 +4997,1741 @@
...
+        "type": "object",
+        "properties": {
+            "relationshipName": {
+                "type": "string",
+                "description": "Relationship Name.",
+                "title": "Relationship Name"
+            },
+            "relatedToCustomer": {
+                "type": "object",
+                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5059]: Please add `Related To Customer -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 35]
```diff
@@ -3903,500 +4997,1741 @@
...
+                    "name": {
+                        "type": "string",
+                        "description": "Name.",
+                        "title": "Name"
+                    },
+                    "addresses": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5089]: Please add `Addresses -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 36]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                    "type": "boolean",
+                                    "description": "Show In Directory."
+                                },
+                                "congressionalDistrict": {
+                                    "type": "string",
+                                    "description": "Congressional District."
+                                },
+                                "countryData": {
+                                    "type": "object",
+                                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5177]: Please add `Country -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 37]
```diff
@@ -3903,500 +4997,1741 @@
...
+                    "imageUri": {
+                        "type": "string",
+                        "description": "Image Uri.",
+                        "title": "Image Uri"
+                    },
+                    "emails": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5237]: Please add `Emails -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 38]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                }
+                            }
+                        },
+                        "description": "Emails."
+                    },
+                    "phones": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5267]: Please add `Phones -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 39]
```diff
@@ -3903,500 +4997,1741 @@
...
+                    "webSite": {
+                        "type": "string",
+                        "description": "Web Site.",
+                        "title": "Web Site"
+                    },
+                    "memberships": {
+                        "type": "array",
+                        "items": {
+        "type": "object",
+        "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Memberships -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 40]
```diff
@@ -3903,500 +4997,1741 @@
...
+                        "description": "Category.",
+                        "title": "Category"
+                    },
+                    "categories": {
+                        "type": "array",
+                        "items": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add `Categories -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 41]
```diff
@@ -3903,500 +4997,1741 @@
...
+        "type": "object",
+        "properties": {
+            "individualId": {
+                "type": "string",
+                "description": "Individual ID.",
+                "title": "Individual ID"
+            },
+            "event": {
+                "type": "object",
+                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5567]: Please add `Event -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 42]
```diff
@@ -3903,500 +4997,1741 @@
...
+                    "attendedDate": {
+                        "type": "string",
+                        "description": "Attended Date.",
+                        "title": "Attended Date"
+                    },
+                    "locations": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5640]: Please add `Locations -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 43]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                },
+                                "description": {
+                                    "type": "string",
+                                    "description": "Description"
+                                },
+                                "addresses": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "object",
+                                        "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5655]: Please add `Addresses -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 44]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                                "type": "boolean",
+                                                "description": "Show In Directory."
+                                            },
+                                            "congressionalDistrict": {
+                                                "type": "string",
+                                                "description": "Congressional District."
+                                            },
+                                            "countryData": {
+                                                "type": "object",
+                                                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5745]: Please add `Country -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 45]
```diff
@@ -3903,500 +4997,1741 @@
...
+                    "registrationDate": {
+                        "type": "string",
+                        "description": "Registration Date.",
+                        "title": "Registration Date"
+                    },
+                    "customFields": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5810]: Please add `Custom Fields -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 46]
```diff
@@ -3903,500 +4997,1741 @@
...
+                        "description": "Custom Fields."
+                    }
+                },
+                "description": "Event."
+            },
+            "sessions": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5838]: Please add `Sessions -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 47]
```diff
@@ -3903,500 +4997,1741 @@
...
+            "badgeState": {
+                "type": "string",
+                "description": "Badge State.",
+                "title": "Badge State"
+            },
+            "boughtTogetherWith": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5948]: Please add `Bought Together With -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 48]
```diff
@@ -3903,500 +4997,1741 @@
...
+                        "quantity": {
+                            "type": "integer",
+                            "format": "int32",
+                            "description": "Quantity."
+                        },
+                        "itemizedCustomFields": {
+                            "type": "array",
+                            "items": {
+                              "type": "object",
+                              "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5989]: Please add `Itemized Custom Fields -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 49]
```diff
@@ -3903,500 +4997,1741 @@
...
+                                "caption": {"type": "string", "description": "Caption".},
+                                "value": {"type": "string", "description": "Value."}
+                                     },
+                            "description": "Itemized Custom Fields."
+                        },
+                        "productCategories": {
+                            "type": "array",
+                            "items": {  
+                              "type": "object",
+                              "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6002]: Please add `Product Categories -` prefix and missing `title`s

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 50]
```diff
@@ -3903,500 +4997,1741 @@
...
+                "title": "Registration Number"
+            }
         }
+    },
+    "CommitteeMemberData": {
+        "type": "object",
+        "properties": {
+            "committee": { 
+                "type": "object",
+                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6069]: There was a typo in my last comment, please replace `Committe` with `Committee`

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 51]
```diff
@@ -3903,500 +4997,1741 @@
...
+            "membershipUniqueId": {
+                "type": "string",
+                "description": "Membership Unique ID.",
+                "title": "Membership Unique ID"
+            },
+            "termsList": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6282]: Please add `Terms -` prefix

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 52]
```diff
@@ -3903,500 +4997,1741 @@
...
+            "categories": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "subCategories": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6556]: Please update prefix

**Review by lordlydon — CHANGES_REQUESTED** [review]
Hi,
            
Please keep in mind that the format for the description we are using is to add the property's name for nested arrays and objects, as there are many cases where both the root and nested objects have fields with the same name.

For example, `Code` and `Name` are properties that appear multiple times in a single response, and it can be confusing when trying to select the correct one in the Power Automate Flow Editor.
If the description shows `Code` and `Education Credits - Code`, it's easier to select the right option.

Another thing is that you seem to be updating only parts of the file, I would recommend copy & pasting the whole file to avoid problems with missing commas or duplicated lines.
And if you try to format the file and some indentations aren't fixed, it's a good sign that there is something missing/extra in the file.

File: certified-connectors/Impexium/apiDefinition.swagger.json (hunk 52 above)
Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json:6556]: Please update the prefix here to `Subcategories -`

**Review by lordlydon — CHANGES_REQUESTED** [review]
There are still some closing brackets missing, I'm getting this error when running paconn validate `Expecting ',' delimiter: line 13574 column 1 (char 525295)`

**Review by lordlydon — CHANGES_REQUESTED** [review]
I'm still getting the same error from paconn validate:
`Expecting ',' delimiter: line 13575 column 1 (char 526501)`

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 53]
```diff
@@ -3929,474 +5023,1855 @@
...
+                    "description": "Committee - Tags.",
+                    "items": {
+                      "type": "string",
+                      "title": "Tag",
+                      "description": "Committee - Tags - Tag."
+                    }
+                 }
+               }
+             },
+            "position": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6169]: Please add missing title

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 54]
```diff
@@ -3929,474 +5023,1855 @@
...
+                "type": "string",
+                "description": "Position."
+            },
+            "positionCode": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6174]: Please add missing title

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 55]
```diff
@@ -3929,474 +5023,1855 @@
...
+                "type": "string",
+                "description": "Position Code."
+            },
+            "code": {
+                "type": "string",
+                "description": "Code.",
+                "title": "Code"
+            },
+            "startDate": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:6184]: Please add missing title

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 56]
```diff
@@ -16,185 +16,834 @@
...
+                "200": {
+                    "description": "default",
+                    "schema": {
+                        "type": "object",
+                        "properties": {
+                            "pageNumber": {
+                                "type": "integer",
+                                "format": "int32",
+                                "description": "Page Number",
+                                "title": ""
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please add missing title

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 57]
```diff
@@ -3929,474 +5023,1855 @@
...
+                    }
                 },
-                "secret": {
-                  "type": "string",
-                  "description": "Secret.",
-                  "title": "Secret",
-                  "x-ms-visibility": "internal",
-                  "default": "Impexium"
+                "description": "Event.",
+                "title": "Event"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please remove the title here as it's added to the name of the children properties.

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 58]
```diff
@@ -3929,474 +5023,1855 @@
...
+                        "title": "Links"
+                    },
+                    "oldId": {
+                        "type": "string",
+                        "description": "Related To Customer - Old ID.",
+                        "title": "Old ID"
+                    }
+                },
+                "description": "Related To Customer.",
+                "title": "Related To Customer"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please remove title here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 59]
```diff
@@ -16,185 +16,834 @@
...
+                            },
+                            "address": {
+                                "type": "string",
+                                "description": "Address.",
+                                "title": "Address"
+                            },
+                            "type": {
+                                "type": "string",
+                                "description": "Type.",
+                                "title": "Address"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please fix title here

**Review by lordlydon — CHANGES_REQUESTED** [review]
The paconn errors are now fixed, and I could update the test connector. Checking the actions in the flow editor I found a couple of other issues, please check the comments.

**Comment by RjImpx** [comment]
> Hi,
> 
> Please keep in mind that the format for the description we are using is to add the property's name for nested arrays and objects, as there are many cases where both the root and nested objects have fields with the same name.
> 
> For example, `Code` and `Name` are properties that appear multiple times in a single response, and it can be confusing when trying to select the correct one in the Power Automate Flow Editor. If the description shows `Code` and `Education Credits - Code`, it's easier to select the right option.
> 
> Another thing is that you seem to be updating only parts of the file, I would recommend copy & pasting the whole file to avoid problems with missing commas or duplicated lines. And if you try to format the file and some indentations aren't fixed, it's a good sign that there is something missing/extra in the file.

@jokano12 please make sure you submit PRs **ONLY** after running paconn validate

**Review by lordlydon — CHANGES_REQUESTED** [review]
I'm getting this error when I run paconn validate.
`Expecting property name enclosed in double quotes: line 5530 column 13 (char 217411)`

**Review by lordlydon — APPROVED** [review]

**Pull request merged**